contribs/gmf/src/directives/layertree.js

```diff
@@ -0,0 +1,296 @@
+   * @private
+   */
+  this.styleStates_ = {};
+
+  //FIXME Choose a theme in gmf datasource
+  $scope.$watch('themes', goog.bind(function(newVal) {
+    if (goog.isDef(newVal)) {
+      this['tree'] = newVal[3];
+    }
+  }, this));
+};
+
+
+/**
+ * Create a return a layer corresponding to the ngeo.layerTree's node.
+ * Can currently only creates WMS layers (internal or external) and WMTS layers.
+ * @param {GmfThemesNode} node Layer tree node.
+ * @return {ol.layer.Layer} The OpenLayers layer or group for the node.
+ * @export
+ */
+gmf.LayertreeController.prototype.getLayer = function(node) {
+  var wmsUrl = this.wmsUrl_;
+  var layerMgr = this.layerManager_;
+
+  var makeLayers = function(node) {
+    var i;
+    var children = node.children;
+    var layers = new ol.Collection();
+    if (goog.isDef(children)) {
+      for (i = 0; i < children.length; i++) {
+        layers.push(makeLayers(children[i]));
+      }
+      return layerMgr.createBasicGroup(layers);
+    }
+    if (node.type === 'external WMS') {
+      return layerMgr.createBasicWMSLayer(node.url, node.name);
+    } else if (node.type === 'WMTS') {
+      return layerMgr.createWMTSLayerFromCapabilitites(node.url, node.name);
+    } else {
+      return layerMgr.createBasicWMSLayer(wmsUrl, node.name);
+    }
+  };
+
+  return makeLayers(node);
+};
+
+
+/**
+ * Return a Gettersetter function used to set style on each line of the tree.
+ * @param {ngeo.LayertreeController} treeCtrl Layer tree controller.
+ * @return {function()} Gettersetter function used as Angular model.
+ * @export
+ */
+gmf.LayertreeController.prototype.getSetStyle =
+    function(treeCtrl) {
+  var map = this.map;
+  var treeCtrlUid = treeCtrl.uid.toString();
+  /** @type {function()} */
+  var styleState = this.styleStates_[treeCtrlUid];
+  if (!goog.isDef(styleState)) {
+    styleState = function() {
+      return gmf.LayertreeController.getSetStyle_(treeCtrl, map);
+    };
+    this.styleStates_[treeCtrlUid] = styleState;
+  }
+  return styleState;
+};
+
+
+/**
+ * Set style of the span element corresponding to the current tree node
+ * Current styles:
+ *  - Add a 'light' class on nodes out of the current map resolution.
+ * @param {ngeo.LayertreeController} treeCtrl Layer tree controller.
+ * @param {ol.Map} map
+ * @private
+ */
+gmf.LayertreeController.getSetStyle_ = function(treeCtrl, map) {
+  var possibleClasses = ['light'];
+  var classes = [];
+  var spans = treeCtrl.element.find('span');
+  var span = angular.element(spans[0]);
+
+  //Add class 'light' when out of resolution;
+  var resolution = map.getView().getResolution();
+  var maxExtent = treeCtrl.node['maxResolutionHint'];
+  var minExtent = treeCtrl.node['minResolutionHint'];
+  if (goog.isDef(minExtent) && resolution < minExtent ||
+      goog.isDef(maxExtent) && resolution > maxExtent) {
+    classes.push(possibleClasses[0]);
+  }
+
+  //update classes on span;
+  span.removeClass(possibleClasses.join(' '));
```

Comment:
Yep. I don't see any documentation or example on how to use a GetterSetter function as model in a ``ngClass``. I think that's simply not possible. If we want to use a ``ngClass``, we need to set another model. Like a property in the node. Then the we would have a attribute like this: ``ngClass="{{::layertreeCtrl.node.style}}"``. I'm not sur that's better... I don't know...